components/centraldashboard/app/api_contributors.ts

```diff
@@ -32,13 +98,30 @@ const EMAIL_RGX = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
 const surfaceProfileControllerErrors = (info: {res: Response, msg: string, err: any}) => {
     const {res, msg, err} = info;
     const code = (err.response && err.response.statusCode) || 400;
-    const error = err.body || msg;
-    console.error(`${msg} ${error}`);
-    apiError({res, code, error});
+    const devError = err.body || '';
+    // Msg is the developer reason of what happened, devError is the technical details as to why
+    console.error(msg+(devError?` ${devError}`:''));
+    apiError({res, code, error: devError || msg});
 };
 
 export class ContributorAPI {
     constructor(private profilesService: DefaultApi) {}
+    /**
+     * Retrieves WorkgroupInfo from Profile Controller for the given user.
+     */
+    static async getWorkgroupInfo(profilesService: DefaultApi, user: User.User): Promise<WorkgroupInfo> {
```

Comment:
Why make this a static method that needs to have the `profilesService` passed into it? If it's a normal instance method `profilesService` is already a private member. It also looks like the signature could just take the user's email as the second parameter.

Reply:
Please refer to [api.ts:55](https://github.com/kubeflow/kubeflow/pull/3758/files#diff-d5346240aaa21ec8e07782abf283707aR59)